libs/providers/config-cat/src/lib/config-cat-provider.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  EvaluationContext,
+  Provider,
+  JsonValue,
+  ResolutionDetails,
+  ParseError,
+  TypeMismatchError,
+  StandardResolutionReasons,
+  ResolutionReason,
+} from '@openfeature/js-sdk';
+import * as configcat from 'configcat-js';
+import { SettingValue } from 'configcat-common/lib/RolloutEvaluator';
+import { IConfigCatClient } from 'configcat-js';
+import { transformContext } from './context-transformer';
+
+export class ConfigCatProvider implements Provider {
+  private client: IConfigCatClient;
+
+  private constructor(client: IConfigCatClient) {
+    this.client = client;
+  }
+
+  public static create(...params: Parameters<typeof configcat.getClient>) {
+    return new ConfigCatProvider(configcat.getClient(...params));
+  }
+
+  public static createFromClient(client: IConfigCatClient) {
+    return new ConfigCatProvider(client);
+  }
+
+  metadata = {
+    name: ConfigCatProvider.name,
+  };
+
+  async resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<boolean>> {
+    const { value, ...evaluationData } = await this.client.getValueDetailsAsync<SettingValue>(
+      flagKey,
+      undefined,
+      transformContext(context)
+    );
+
+    if (typeof value !== 'undefined' && !(typeof value === 'boolean')) {
+      throw new TypeMismatchError(`Requested boolean flag but the actual value is ${typeof value}`);
+    }
```

Comment:
Nitpick: you could possibly extract this similar logic into a util function, where the expected type is a parameter, not required though.

Reply:
Yeah thats a good point, I added it.